test/integration/framework/utils.go

```diff
@@ -0,0 +1,309 @@
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+						},
+					},
+					HostNetwork:  opts.EnableHostNetwork,
+					NodeSelector: nodeSelectors,
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+// getFirstPodWithLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getFirstPodWithLabels(ctx context.Context, labelsMap map[string]string, namespace string, client kubernetes.Interface) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
+
+	podList, err := getPodsByLabels(ctx, labelsMap, client, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		pod := podList.Items[0]
+		runningPodWithDesiredLabels = &pod
+	}
+	return runningPodWithDesiredLabels, nil
+}
+
+// getFirstRunningPodWithLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getFirstRunningPodWithLabels(ctx context.Context, labelsMap map[string]string, seedOrShoot ClusterType) (*corev1.Pod, error) {
+	var (
+		podList *corev1.PodList
+		err     error
+	)
+	switch seedOrShoot {
+	case Seed:
+		podList, err = getPodsByLabels(ctx, labelsMap, o.K8sSeedClient, o.Shoot.SeedNamespace)
+		if err != nil {
+			return nil, err
+		}
+	case Shoot:
+		podList, err = getPodsByLabels(ctx, labelsMap, o.K8sShootClient, metav1.NamespaceSystem)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(podList.Items) == 0 {
+		return nil, ErrNoRunningPodsFound
+	}
+
+	var runningPod *corev1.Pod
+	for _, pod := range podList.Items {
+		o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+		if pod.Status.Phase == corev1.PodRunning {
+			runningPod = &pod
+			break
+		}
+	}
+
+	return runningPod, nil
+}
+
+// GetAdminPassword gets the admin password for authenticating agains the api
+func (o *GardenerTestOperation) getAdminPassword(ctx context.Context) (*string, error) {
+	password, err := getObjectFromSecret(ctx, o.K8sSeedClient, o.Shoot.SeedNamespace, kubecfg, password)
+	return &password, err
+}
+
+func (s *ShootGardenerTest) mergePatch(ctx context.Context, oldShoot, newShoot *v1beta1.Shoot) error {
+	patchBytes, err := kubernetesutils.CreateTwoWayMergePatch(oldShoot, newShoot)
+	if err != nil {
+		return fmt.Errorf("failed to patch bytes")
+	}
+
+	_, err = s.K8sGardenClient.Garden().GardenV1beta1().Shoots(s.Shoot.Namespace).Patch(s.Shoot.Name, types.StrategicMergePatchType, patchBytes)
+	if err != nil {
+		return fmt.Errorf("failed to update shoot with new value: %s", err.Error())
+	}
+
+	return nil
+}
+
+func getPodsByLabels(ctx context.Context, labelsMap map[string]string, client kubernetes.Interface, namespace string) (*corev1.PodList, error) {
```

Comment:
You should use a `labels.Selector` instead of passing a `labelsMap` around, then in the future if you have more refined criteria to search for a pod you don't need to adapt this core function anymore.

Reply:
What about this comment?

Reply:
resolved